.eslintrc.js

```diff
@@ -127,4 +78,55 @@ module.exports = {
     eqeqeq: 'error',
     // keep-sorted end
   },
+  overrides: [
+    {
+      files: ['*.ts'],
+      extends: [
+        'plugin:@typescript-eslint/eslint-recommended',
+        'plugin:@typescript-eslint/recommended',
+        'plugin:@typescript-eslint/recommended-requiring-type-checking',
+      ],
+      parserOptions: {
+        project: './tsconfig.base.json',
```

Comment:
This contains all the configuration that the parser needs, that way we can use TS (if needed) outside of the project and EsLint will not complain.